packages/amplify-cli/src/domain/amplify-toolkit.ts

```diff
@@ -461,4 +456,9 @@ export class AmplifyToolkit {
   getCleanUpTasks(): Function[] {
     return this._cleanUpTasks;
   }
+
+  async runCleanUpTasks(context: Context) {
```

Comment:
need to use `async runCleanUpTasks: () => { }` notation. Otherwise `this` is bound to the calling context rather than the semantic context